### ISSUE TITLE
gerbera doesn't restart on SIGHUP

### ISSUE THREAD

**Issue by druban** [gerbera/gerbera#2101]
Gives me this:
```
2021-11-01 10:56:27.099  error: [main.cc:576] main(): Could not restart Gerbera
2021-11-01 10:56:27.099  error: [main.cc:591] main(): main: error [/usr/home/dima/dev/gerbera/src/config/config_manager.cc:530] getOption Error: option 19 not set
2021-11-01 10:56:27.099   info: [main.cc:604] main(): Gerbera exiting. Have a nice day.
```

**Comment by cweiske** [comment]
SIGHUP handling still has problems with v2.1.0-59-gb4818f18

At first it tries to restart, but the UPnP part fails:
```
# killall -SIGHUP gerbera
```

```
gerbera[1384799]: 2024-06-05 07:52:08   info: Restarting Gerbera!
gerbera[1384799]: 2024-06-05 07:52:08   info: Loading configuration from: /etc/gerbera/config.xml
gerbera[1384799]: 2024-06-05 07:52:08   info: Parsing configuration...
[...]
gerbera[1384799]: 2024-06-05 07:52:38  error: Could not restart Gerbera
gerbera[1384799]: 2024-06-05 07:52:38  error: UpnpUnRegisterClient failed (-100)
gerbera[1384799]: 2024-06-05 07:52:38  error: UpnpUnRegisterRootDevice failed (-100)
```

**Comment by KarlStraussberger** [comment]
I refactored the main code and cannot reproduce the issue anymore (#3118). Maybe you are once again so kind and take a look at it 😃

**Comment by cweiske** [comment]
When using cd618bc I have problems:
- First SIGHUP worked
- After the second SIGHUP, VLC did not find the UPnP server anymore
- Third had the same problem

Restarting the gerbera service via systemctl made it reappear in VLC.

(Also, I get unnecessary warnings for added inotify watches in the log)

**Comment by KarlStraussberger** [comment]
From the analysis and the fix this issue was persisting since the introduction of pthread library.

**Comment by cweiske** [comment]
.. which was 2006 or earlier: 393f17d9d3bf01a2b01a25aa43b0744c442a9c16 :)

**Comment by KarlStraussberger** [comment]
closed by #3118
